src/providers/google/helpers.ts

```diff
@@ -171,3 +265,296 @@ export class GmailHelpers {
     return { name: "", email: "" };
   }
 }
+
+export class GoogleDriveHelpers {
+  static async getFile(
+    drive: drive_v3.Drive,
+    fileId: string
+  ): Promise<drive_v3.Schema$File> {
+    try {
+      const res = await drive.files.get({
+        fileId: fileId,
+        fields: 'id, name, mimeType, size, webViewLink, createdTime, modifiedTime, thumbnailLink'
+      });
+      return res.data;
+    } catch (error) {
+      console.error("Error getting file:", error);
+      throw error;
+    }
+  }
+
+  static async getFileSize(
+    drive: drive_v3.Drive,
+    fileId: string
+  ): Promise<number | undefined> {
+    const file = await this.getFile(drive, fileId);
+
+    if (file.size) {
+      // For non-Google docs (like PDF, image)
+      return parseInt(file.size);
+    } else if (file.mimeType && file.mimeType.startsWith("application/vnd.google-apps.")) {
+      // For Google Docs, export the file as plain text to estimate size
+      const exportedFile = await drive.files.export(
+        { fileId: fileId, mimeType: "text/plain" }, 
+        { responseType: "arraybuffer" }
+      );
+      return Buffer.byteLength(exportedFile.data as ArrayBuffer);
+    } else {
+      return undefined;
+    }
+  }
+
+  static async extractTextContent(
+    drive: drive_v3.Drive,
+    fileId: string,
+    mimeType: string,
+    auth: OAuth2Client
+  ): Promise<string> {
+    let textContent = "";
+
+    try {
+      // Attempt to fetch the indexable text from contentHints
+      const response = await drive.files.get({
+        fileId: fileId,
+        fields: "contentHints/indexableText, size",
+      });
+
+      // Check if indexable text is available
+      textContent = response.data.contentHints?.indexableText || "";
+
+      // If no indexable text is found, proceed with original extraction methods
+      if (!textContent) {
+        console.warn("No indexable text found, using fallback method.");
+
+        // 10MB limit (5 * 1024 * 1024)
+        const sizeLimit = 10 * 1024 * 1024;
```

Comment:
Set this as a config option in the google config.

Reply:
> Set this as a config option in the google config.

Done